torchvision/datasets/phototour.py

```diff
@@ -0,0 +1,178 @@
+import os
+import numpy as np
+
+import cv2
+
+import torch
+import torch.utils.data as data
+
+
+class PhotoTour(data.Dataset):
+    urls = {
+        'notredame': 'http://www.iis.ee.ic.ac.uk/~vbalnt/phototourism-patches/notredame.zip',
+        'yosemite': 'http://www.iis.ee.ic.ac.uk/~vbalnt/phototourism-patches/yosemite.zip',
+        'liberty': 'http://www.iis.ee.ic.ac.uk/~vbalnt/phototourism-patches/liberty.zip'
+    }
+    mean = {'notredame': 0.4854, 'yosemite': 0.4844, 'liberty': 0.4437}
+    std = {'notredame': 0.1864, 'yosemite': 0.1818, 'liberty': 0.2019}
+    lens = {'notredame': 468159, 'yosemite': 633587, 'liberty': 450092}
+
+    image_ext = 'bmp'
+    info_file = 'info.txt'
+    matches_files = 'm50_100000_100000_0.txt'
+
+    def __init__(self, root, name, download=False, size=None):
+        self.root = root
+        self.size = size or 64
+
+        self.name = name
+        self.data_dir = os.path.join(root, name)
+        self.data_down = os.path.join(root, '{}.zip'.format(name))
+        self.data_file = os.path.join(root, '{}_{}.pt'.format(name, self.size))
+
+        self.mean = self.mean[name]
+        self.std = self.std[name]
+
+        if download:
+            self.download()
+
+        if not self._check_exists():
+            raise RuntimeError('Dataset not found.'
+                               + ' You can use download=True to download it')
+
+        # load the serialized data
+        self.data, self.labels, self.matches = torch.load(self.data_file)
+
+    def __getitem__(self, index):
+        return self.data[index]
+
+    def __len__(self):
+        return self.lens[self.name]
+
+    def _check_exists(self):
+        return os.path.exists(self.data_file)
+
+    def _check_downloaded(self):
+        return os.path.exists(self.data_dir)
+
+    def download(self):
+        print_('\n-- Loading PhotoTour dataset: {}\n'.format(self.name))
+
+        if self._check_exists():
+            print_('# Found cached data {}'.format(self.data_file))
+            return
+
+        if not self._check_downloaded():
+            # download files
+            url = self.urls[self.name]
+            filename = url.rpartition('/')[2]
+            file_path = os.path.join(self.root, filename)
+
+            try:
+                os.makedirs(self.root)
+            except Exception:
+                    pass
+
+            print_('# Downloading {}\n# Downloading {}\n\nIt might take while.'
+                   ' Please grab yourself a coffee and relax.\n'
+                   .format(url, file_path))
+
+            os.system('wget {} -P {}'.format(url, self.root))
```

Comment:
Not all systems have `wget` installed by default. Use `urllib`. See how it's done in the MNIST dataset.